### DISCUSSION TITLE
support manually search button

### DISCUSSION THREAD

**Discussion by imiskolee** [directus/directus#12251] [Feature Requests]
hi:

I know we are using automatically search function solution, but if working on a big table it's not friendly with database, because of：
   the API just channeled on the frontend, backend still running it.
 
how can we support a manual search button?

**Comment by benhaynes** [comment]
Sorry, I don't quite understand. Could you explain a bit more?

**Reply by imiskolee** [reply]
> Sorry, I don't quite understand. Could you explain a bit more?

when we text on search text input, will call API on each change. For example, if we want to search `FREEDOM`, the application will be searched 7 times. if the backend is too slow(search on 10M records table), automatically search very bad.

**Reply by rijkvanzanten** [reply]
Right now it searches with a throttle on the input. That normally doesn't cause too many issues, but if the actual search takes a while (like when searching on a huge data set) then yes that would cause some less than ideal experiences

**Reply by benhaynes** [reply]
I think the ideal solution here is adding proper search alternatives, per your other ticket: https://github.com/directus/directus/discussions/12264

**Reply by imiskolee** [reply]
> I think the ideal solution here is adding proper search alternatives, per your other ticket: #12264

yes, we also tested all databases (included multi-node cockroanchDB), like search also too bad.

**Reply by LevelUpGit** [reply]
We noticed same issue.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
